desc/equilibrium/coords.py

```diff
@@ -98,27 +98,32 @@ def map_coordinates(  # noqa: C901
     kwargs.setdefault("tol", tol)
     kwargs.setdefault("maxiter", maxiter)
     period = np.asarray(period)
-    coords = coords % period
+
+    def periodic(x):
+        return jnp.where(jnp.isfinite(period), x % period, x)
+
+    coords = periodic(coords)
 
     params = setdefault(params, eq.params_dict)
 
+    profiles = get_profiles(inbasis + basis_derivs, eq, None)
+    # do surface average to get iota once
+    if "iota" in profiles and profiles["iota"] is None:
+        profiles["iota"] = eq.get_profile("iota", params=params)
+
     @functools.partial(jit, static_argnums=1)
     def compute(y, basis):
         grid = Grid(y, sort=False, jitable=True)
-        profiles = get_profiles(inbasis + basis_derivs, eq, grid, jitable=True)
-        # do surface average to get iota once
-        if "current" in profiles and profiles["current"] is not None:
-            profiles["iota"] = eq.get_profile("iota")
-        transforms = get_transforms(inbasis + basis_derivs, eq, grid, jitable=True)
+        transforms = get_transforms(basis, eq, grid, jitable=True)
         data = compute_fun(eq, basis, params, transforms, profiles)
         x = jnp.array([data[k] for k in basis]).T
         return x
 
     @jit
     def residual(y, coords):
         xk = compute(y, inbasis)
-        r = xk % period - coords % period
-        return jnp.where(r > period / 2, -period + r, r)
+        r = periodic(xk) - periodic(coords)
+        return jnp.where((r > period / 2) & jnp.isfinite(period), -period + r, r)
```

Comment:
how did this change fix the function for negative values? I don't quite follow

Reply:
we're using `period=np.inf` to denote no period, which is fine for quantities that are always positive (like `R`) because `1.23 % np.inf == 1.23`. However, `-1.23 % np.inf == np.inf` so it breaks for quantities that can be negative (like `Z`). This only does the modulo division where period is finite, which fixes the issue.